tests/Entity/PageRevisionTest.php

```diff
@@ -29,4 +28,30 @@ public function test_revision_count_shown_in_page_meta()
         $pageView->assertSee('Revision #' . $page->revision_count);
     }
 
+    public function test_revision_deletion() {
+        $page = Page::first();
+        $this->asEditor()->put($page->getUrl(), ['name' => 'Updated page', 'html' => 'new page html', 'summary' => 'Update a']);
+        $this->asEditor()->put($page->getUrl(), ['name' => 'Updated page', 'html' => 'new page html', 'summary' => 'Update a']);
+        $page = Page::find($page->id);
+        $beforeRevisionCount = $page->revisions->count();
+
+        // Delete the first revision
+        $revision = $page->revisions->get(0);
+        $resp = $this->asEditor()->delete($revision->getUrl('/delete/'));
+        $resp->assertStatus(200);
+
+        $page = Page::find($page->id);
+        $afterRevisionCount = $page->revisions->count();
+
+        $this->assertTrue($beforeRevisionCount === ($afterRevisionCount + 1));
```

Comment:
Tests are failing here. Took me a while but found why, Actually is a mistake in my own test above this one so my bad :man_facepalming: 

There is only one revision being created. The call to delete does not delete the revision since you've got a check to not delete the latest revision (Which also returns a `200` code as mentioned above).

There is only one revision before on the first update call (Line 33) the page slug updates due to the page name change. Therefore the second update call (Line 34) does not actually update the page since the page slug used (via `$page->getUrl()`) is the old one (Since the test `$page` object is not updated by the change in the application).

To fix this you can simply add a `$page = Page::find($page->id);` between the two update calls.
If you could do the same for my test (`test_revision_count_shown_in_page_meta`) that would be ideal.

Reply:
Argh!! I should have seen that!